From 976413d2362355fdb7b8792ffb7784d7da2ccd24 Mon Sep 17 00:00:00 2001
From: bo huang <bo.b.huang@intel.com>
Date: Mon, 23 Jul 2012 16:39:30 +0800
Subject: [PORT FROM MAIN] MTP/PTP: After disable PTP function while DUT is transfering file via PTP, device can't work again.

BZ: 22039

Handle I/O error when disable PTP function.

Change-Id: Ia4557fea1c645155a2c3439e916a4d6c61a1d5d1
Orig-Change-Id: Iac59f7369978faecc1a658a6b3e0fa660fe89901
Signed-off-by: bo huang <bo.b.huang@intel.com>
Signed-off-by: Wu, Hao <hao.wu@intel.com>
Reviewed-on: http://android.intel.com:8080/76540
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Meng, Zhe <zhe.meng@intel.com>
Tested-by: Meng, Zhe <zhe.meng@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/mtp/MtpServer.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/mtp/MtpServer.cpp b/media/mtp/MtpServer.cpp
index df87db4..3f8e4a7 100644
--- a/media/mtp/MtpServer.cpp
+++ b/media/mtp/MtpServer.cpp
@@ -958,13 +958,13 @@ MtpResponseCode MtpServer::doSendObject() {
         ALOGV("receiving %s\n", (const char *)mSendObjectFilePath);
         // transfer the file
         ret = ioctl(mFD, MTP_RECEIVE_FILE, (unsigned long)&mfr);
-        ALOGV("MTP_RECEIVE_FILE returned %d\n", ret);
+        ALOGV("MTP_RECEIVE_FILE returned %d errno %d \n", ret, errno);
     }
     close(mfr.fd);
 
     if (ret < 0) {
         unlink(mSendObjectFilePath);
-        if (errno == ECANCELED)
+        if (errno == ECANCELED || errno == EIO )
             result = MTP_RESPONSE_TRANSACTION_CANCELLED;
         else
             result = MTP_RESPONSE_GENERAL_ERROR;
-- 
1.7.4.1

